modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,514 @@
+        """
+        Swap the row and column axes.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the row and column axes swapped.
+
+        Notes
+        -----
+        Transposing a dataframe is expensive, and thus, while the axes are swapped
+            logically immediately, the physical swap does not occur until absolutely necessary,
+            which helps motivate the axis argument to the other operators in this algebra.
+        """
+        pass
+
+    @abstractmethod
+    def to_labels(self, column_labels: Union[str, List[str]]) -> "ModinDataframe":
+        """
+        Replace the row labels with one or more columns of data.
+
+        Parameters
+        ----------
+        column_labels : string or list of strings
+            Column label(s) to use as the new row labels.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the row labels replaced by the specified columns.
+
+        Notes
+        -----
+        When multiple column labels are specified, a heirarchical set of labels is created, ordered by the ordering
+            of labels in the input.
+        """
+        pass
+
+    @abstractmethod
+    def from_labels(self) -> "ModinDataframe":
+        """
+        Move the row labels into the data at position 0, and sets the row labels to the positional notation.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the row labels moved into the data.
+
+        Notes
+        -----
+        In the case that the dataframe has hierarchical labels, all label "levels” are inserted into the dataframe
+            in the order they occur in the labels, with the outermost being in position 0.
+        """
+        pass
+
+    @abstractmethod
+    def rename(
+        self,
+        new_row_labels: Optional[Union[Dict[Hashable, Hashable], Callable]] = None,
+        new_col_labels: Optional[Union[Dict[Hashable, Hashable], Callable]] = None,
+        level: Optional[Union[int, List[int]]] = None,
+    ) -> "ModinDataframe":
+        """
+        Replace the row and column labels with the specified new labels.
+
+        Parameters
+        ----------
+        new_row_labels : dictionary or callable, optional
+            Mapping from old row labels to new labels.
+        new_col_labels : dictionary or callable, optional
+            Mapping from old col labels to new labels.
+        level : int or list of ints, optional
+            Level(s) whose row labels to replace.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the new row and column labels.
+
+        Notes
+        -----
+        If level is not specified, the default behavior is to replace row labels in all levels.
+        """
+        pass
+
+    @abstractmethod
+    def sort_by(
+        self,
+        axis: Union[int, Axis],
+        columns: Union[str, List[str]],
```

Comment:
Hmm in this case I think columns works well because its what people are used to seeing, and is fairly explicit. In the other case, the concern was confusion over having two different args correspond to columns, which is why we changed it.